source/common/http/header_map_impl.cc

```diff
@@ -19,10 +19,15 @@ constexpr size_t MinDynamicCapacity{32};
 // This includes the NULL (StringUtil::itoa technically only needs 21).
 constexpr size_t MaxIntegerLength{32};
 
+size_t newCapacity(uint32_t existing_capacity, uint32_t size_to_append) {
```

Comment:
Would it work to make the function args both size_t instead of uint32_t? Then no casts would be needed in the function.

Reply:
+1 on switching everything to explicit `uint32_t` and `uint64_t`. Too hard to reason about as is.

Reply:
I'd mildly prefer the cast over changing inputs, because otherwise one could pass in a real uint64_t and overflow numeric_limits<uint64_t>::max()

I'm claiming having a death test validates that we got the types correct, though I can (and maybe should) add one in header_map_impl_test.cc as well as the user defined header test which motivated this change in the first place.

Reply:
Annnnnd found another overflow bug.
Sigh.
@htuch we're going to fuzz length limits, yeah?

Reply:
Yes, I will put this on my fuzzer roadmap. The issue today is that we fuzz the inputs, but the fuzzer itself doesn't generate massive inputs in terms of GB sized protos. What we need to do is make https://github.com/envoyproxy/envoy/blob/master/test/common/http/header_map_impl_fuzz.proto#L11 a `oneof` with an integer that could just add `std::string(n, 'A')`.